### PR TITLE
Fix The Maker's Eye

### LINKED ISSUE #4864: Doppelgänger doubling The Maker's Eye effect
When the runner plays [The Maker's Eye](https://netrunnerdb.com/en/card/01036) while [Doppelgänger](https://netrunnerdb.com/en/card/20025) is on the field, if the run is successful they can choose to make another run with Doppelganger's effect, if they choose to make this free run on R&D the effect from The Maker's Eye triggers again if the run is successful, allowing them to access three cards from R&D twice.

### PR DESCRIPTION
**Pull request by NoahTheDuke**
A redo of #4866

Closes #4864

### CODE REVIEW

File: src/clj/game/cards/events.clj [hunk 1]
```diff
@@ -2667,6 +2667,7 @@
    :async true
    :effect (effect (make-run eid :rd nil card))
    :events [{:event :successful-run
+             :unregister-once-resolved true
```
Comment (lostgeek) [inline src/clj/game/cards/events.clj]: How is the event triggering when Maker's Eye is in the discard? Shouldn't those cards be inactive?

Comment (NoahTheDuke) [reply src/clj/game/cards/events.clj]: It’s active in the play area and trashed the effect finishes.

Comment (lostgeek) [reply src/clj/game/cards/events.clj]: Sure, but shouldn't it be trashed as soon as the run ends and not be triggered on the second Dopplegänger run?

Comment (NoahTheDuke) [reply src/clj/game/cards/events.clj]: No, cuz the Doppelganger run is created in 10.3.1 and the rest of the steps wait until Doppelganger's run is finished.

Comment (lostgeek) [reply src/clj/game/cards/events.clj]: Ah, so Maker's Eye isn't in the trash during the Doppelgänger run, but it's effect is read as if it said "If _that run_ is successful, access 2 additional cards from R&D". Correct?

Comment (NoahTheDuke) [reply src/clj/game/cards/events.clj]: Yep!

**Pull request merged**